components/infobox/extensions/commons/metadata_generator.lua

```diff
@@ -0,0 +1,172 @@
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local String = require('Module:String')
+local Localisation = require('Module:Localisation')
+local Template = require('Module:Template')
+local Games = require('Module:Games')
+local Variables = require('Module:Variables')
+local StringUtils = require('Module:StringUtils')
+local Class = require('Module:Class')
+
+local MetadataGenerator = {}
+
+function MetadataGenerator.tournament(args)
+	if String.isEmpty(args.publisherdescription) then
+		error('You must provide the publisherdescription param!')
+	end
+
+	local output
+	local frame = mw.getCurrentFrame()
+
+	local name = not String.isEmpty(args.name) and (args.name):gsub('&nbsp;', ' ') or mw.title.getCurrentTitle()
+
+	local type = args.type
+	local locality = Localisation.getLocalisation({displayNoError = true}, args.country)
+
+	local organizers = {
+		args['organizer-name'] or args.organizer,
+		args['organizer2-name'] or args.organizer2,
+		args['organizer3-name'] or args.organizer3,
+		args['organizer4-name'] or args.organizer4,
+		args['organizer5-name'] or args.organizer5
+	}
+
+	local tier = args.liquipediatier and Template.safeExpand(frame, 'TierDisplay', {args.liquipediatier}) or nil
+
+	if tier then
+		tier = (tonumber(
+			Template.safeExpand(frame, 'TierDisplay/number', {args.liquipediatier})
+		) or 0) > 4 and tier:lower() or tier
+	else
+		tier = 'Unknown Tier'
+	end
+
+	local tierType = (tier == 'qualifier' or tier == 'showmatch') and tier or 'tournament'
+	local publisher = Variables.varDefault(args.publisherdescription, '')
+	local date, tense = MetadataGenerator.getDate(args.edate or args.date, args.sdate)
+
+	local teams = args.team_number
+	local players = args.player_number
+
+	local game = (not String.isEmpty(args.game) and args.game ~= 'csgo') and Games.abbr[args.game]
+
+	local prizepoolusd = args.prizepoolusd and ('$' .. args.prizepoolusd .. ' USD') or nil
+	local prizepool = prizepoolusd or (
+		args.prizepool and args.localcurrency and (
+			Variables.varDefault('localcurrencysymbol', '') .. args.prizepool ..
+			Variables.varDefault('localcurrencysymbolafter', '') .. ' ' ..
+			Variables.varDefault('localcurrencycode', '')
+		)
+	)
+	local charity = args.charity == 'true' and true
+	local dateVerb = (tense == 'past' and 'took place ') or (tense == 'future' and 'will take place ') or 'takes place '
+	local dateVerbPublisher =
+		(tense == 'past' and ' which took place ') or
+		(tense == 'future' and ' which will take place ') or
+		' taking place '
+
+	output = StringUtils.interpolate('${name} is a${type}${locality}${game}${charity}${tierType}${organizer}', {
+		name = name,
+		type = type and ('n ' .. type:lower() .. ' ') or '',
+		locality = locality and (locality .. ' ') or '',
+		game = game and (game .. ' ') or '',
+		charity = charity and 'charity ' or '',
+		tierType = tierType,
+		organizer = organizers[1] and (' organized by ' .. organizers[1]) or ''
+	})
```

Comment:
`a` vs `an` can be a problem here if type is not set

Reply:
Added null handler

Reply:
that is not what i meant, should have explained better what i meant^^

was meaning grammar not a possible nil concat

if you do not have type but e.g. locality `English` the sentence would be
`... is a English ...`

(with the null check you now added it will be always `an` so you get the same issue with e.g. locality `German` now --> `... is an German ...`)

basically you would need a function that gets the first used word of type, locality, game, charity, tiertype, organizer and then check for that word if it needs `a` or `an`
(can use https://liquipedia.net/commons/Module:A_or_an for the `a` vs `an` check, documentation via https://liquipedia.net/commons/Template:A_or_an)

Reply:
Done